docs/hooks.md

```diff
@@ -59,7 +59,22 @@ This is used in SystemJS core to ensure that import maps are loaded so that the
 
 This function downloads and executes the code for a module. The promise must resolve with a "register" array, as described in the `getRegister` documentation.
 
-The default system.js implementation is to append a script tag that downloads and executes the module's code, subsequently resolving the promise with the most recent register: `resolve(System.getRegister())`. [Example](https://github.com/systemjs/systemjs/blob/master/src/features/script-load.js).
+The default system.js implementation is to append a script tag that downloads and executes the module's code, subsequently resolving the promise with the most recent register: `resolve(System.getRegister())`. 
+
+It is impossible to have an "identifier" to mark the module identity like AMD therefore SystemJS is using a race condition prone way to register the module. Therefore, you must make sure to maintain the execution order of the code.
+
+The correct execution order looks like this:
+
+```
+async System.instantiate
+  => (The module calls) System.register
+  => System.instantiate noticed the module executed **synchorously** 
```

Comment:
This is actually the synchonous onload even of the script tag.

Reply:
As mentioned, this is incorrect.

Reply:
This part has removed in the new example